samples/bluetooth/mesh/prj_bbc_microbit.conf

```diff
@@ -2,8 +2,17 @@ CONFIG_INIT_STACKS=y
 CONFIG_MAIN_STACK_SIZE=512
 CONFIG_MICROBIT_DISPLAY=y
 CONFIG_SYS_CLOCK_TICKS_PER_SEC=250
+CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=2048
 CONFIG_GPIO=y
 
+CONFIG_BT_SETTINGS=y
+CONFIG_FLASH=y
+CONFIG_FLASH_PAGE_LAYOUT=y
+CONFIG_FLASH_MAP=y
+CONFIG_FCB=y
+CONFIG_SETTINGS=y
```

Comment:
Shouldn't this be selected automatically by `CONFIG_BT_SETTINGS`?

Reply:
Btw, this is a copy from prj.conf that was already merged, so both places will need fixing if I switch to `select`. FWIW, I'm fine with `select` in this case - I was just wondering if we have documented reasoning to guide developers with choosing between `depends on` and `select`.

Reply:
Now this confuses me. I have been told (can't remember by whom) to always use `select` and not `depends on` so as to avoid the number of lines in a `.conf`. Remember that the difference between Linux and Zephyr is `oldconfig` and using `.config` files directly. In Zephyr `.config` files are treated right now purely as output build files, not as files you keep for your app. I would say that is the reason why `selects` is a better choice for Zephyr.

Reply:
@SebastianBoe @ulfalizer might want to weigh in

Reply:
I think the general idea is that you don't end up getting unexpected features into your build, and you don't end up getting conflicts where you're requested a feature but it's trying to select another features who's pre-requisites aren't fulfilled (it might not be available for your board, you don't have some third feature enabled, or something else). I.e. in summary try to eliminate too much implicit/magic stuff happening behind the scenes which you can't see or control very well.

Reply:
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/kbuild/kconfig-language.txt#n128

Reply:
Btw, based on the above reasoning in kconfig-language.txt, BT_SETTINGS should use `depends on` since SETTINGS has its own dependencies on NFFS || FCB

Reply:
Ahh... I see how `depends on` came up now.

One case where you should avoid `select` is for simple dependencies, like below:

```
config FOO_DEP
    bool "foo_dep"

config FOO
    bool "foo"
    select FOO_DEP
```

This is cleaner:

```
config FOO_DEP
    bool "foo_dep"

config FOO
    bool "foo"
    depends on FOO_DEP
```

`select` makes dependencies harder to follow, since you can't tell from looking at a symbol that other things select it. Often you run into dependency issues too. `select` should mostly be restricted to "helper" symbols.

As a bonus, `FOO` will get indented below `FOO_DEP` there if you use `depends on`.

Reply:
OK, thanks @ulfalizer for the input. Let's go with `select` then.

Reply:
> OK, thanks @ulfalizer for the input. Let's go with select then.

@carlescufi I think you mean `depends on`? Since SETTINGS is not a "helper" symbol.

Reply:
oops, yes, I meant `depends on`. Apologies.